core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -1558,7 +1558,7 @@ private[spark] object JsonProtocol {
     json.extractElements.map { line =>
       val declaringClass = line.get("Declaring Class").extractString
       val methodName = line.get("Method Name").extractString
-      val fileName = line.get("File Name").extractString
+      val fileName = jsonOption(line.get("File Name")).map(_.extractString).orNull
```

Comment:
Yes, we cannot it will be non-null.
Also handle exception.message below ?

Reply:
I'll send separate PR to handle `exception.message`